pkg/controller/machine/bootstrap.go

```diff
@@ -238,7 +247,15 @@ fi
 {{- if .EnterpriseLinux }}
 yum install epel-release -y
 {{- end }}
+
+{{- if and (eq .ProviderSpec.CloudProvider "azure") (eq .ProviderSpec.OperatingSystem "rhel") }}
+yum --disablerepo='*' remove -y 'rhui-azure-rhel8'
+wget https://rhelimage.blob.core.windows.net/repositories/rhui-microsoft-azure-rhel8.config
+yum --config=rhui-microsoft-azure-rhel8.config install -y rhui-azure-rhel8
+{{- end }}
```

Comment:
This seems to enable Extended Update Support (EUS) as per https://docs.microsoft.com/en-us/azure/virtual-machines/workloads/redhat/redhat-rhui, which seems to be only necessary if you want to pin to a specific minor release. Why are we adding it?

Reply:
Plus, will this even work on images that do "bring your own license"?

Reply:
Actually it is not related to the EUS, I used a different config from the one mentioned in the post (it misses the -eus suffix). It was needed to make the whole thing work as yum wasn't able to fetch metadata packages. This is the only way that I found, removing the repo and adding it again

Reply:
Ah, fair point. Did not notice that. I'm somewhat confused by that though.

Any idea if we support https://docs.microsoft.com/en-us/azure/virtual-machines/workloads/redhat/byos? I'm not very familiar with RHEL on Azure. Maybe we can improve this snippet by only running it when yum has that specific repo configured?

Reply:
@mlavacca what was the error that you were getting ?

Reply:
@moadqassem 
```
sudo yum install curl
Updating Subscription Management repositories.
Red Hat Enterprise Linux 8 for x86_64 - BaseOS from RHUI (RPMs)                  0.0  B/s |   0  B     00:00     Errors during downloading metadata for repository 'rhui-rhel-8-for-x86_64-baseos-rhui-rpms':
- Curl error (56): Failure when receiving data from the peer for https://rhui-3.microsoft.com/pulp/repos/content/dist/rhel8/rhui/8/x86_64/baseos/os/repodata/repomd.xml [OpenSSL SSL_read: error:14094415:SSL routines:ssl3_read_bytes:sslv3 alert certificate expired, errno 0]
- Curl error (56): Failure when receiving data from the peer for https://rhui-1.microsoft.com/pulp/repos/content/dist/rhel8/rhui/8/x86_64/baseos/os/repodata/repomd.xml [OpenSSL SSL_read: error:14094415:SSL routines:ssl3_read_bytes:sslv3 alert certificate expired, errno 0]
- Curl error (56): Failure when receiving data from the peer for https://rhui-2.microsoft.com/pulp/repos/content/dist/rhel8/rhui/8/x86_64/baseos/os/repodata/repomd.xml [OpenSSL SSL_read: error:14094415:SSL routines:ssl3_read_bytes:sslv3 alert certificate expired, errno 0]
- Error: Failed to download metadata for repo 'rhui-rhel-8-for-x86_64-baseos-rhui-rpms': Cannot download repomd.xml: Cannot download repodata/repomd.xml: All mirrors were tried
```

Reply:
Is it possible that the problem is that the default RHEL image on Azure is BYOS, not RHUI?

https://github.com/kubermatic/machine-controller/blob/fce1320fd139f6ee48f762a2e8696f02aa87ed0c/pkg/cloudprovider/provider/azure/provider.go#L142-L147

Reply:
(that RHEL version is also fairly old; maybe we should bump it?)

Reply:
Should go to 8.5 at least